container/config.go

```diff
@@ -0,0 +1,178 @@
+package container
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"reflect"
+
+	"github.com/pkg/errors"
+
+	"github.com/goccy/go-graphviz"
+	"github.com/goccy/go-graphviz/cgraph"
+)
+
+type config struct {
+	autoGroupTypes   map[reflect.Type]bool
+	onePerScopeTypes map[reflect.Type]bool
+
+	// logging
```

Comment:
I would extract the logging and the graphing from the options struct. I see the config used only to configure the container. As it is done when constructing with autoGroupTypes and onePerScope types.

Reply:
Where I would put them instead?

Reply:
The alternative would be adding a parameter or overload to `Run` just with debugging options separate from the container options. But that feels relatively involved. If you think it's useful to separate how about we align on an API and do that in a follow-up PR?